workstation-config/securedrop-mime-handling

```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
+##
+# securedrop-mime-handling
+# =====================
+#
+# Overrides mimetype handling for certain VMs. Instead of relying on the
+# /usr/share/applications (system volume), we instead use /home/user/.local/share/
+# to be provisioned on boot by systemd.
+##
+
+# Fail early to ensure all the script ran successful
+set -e
+
+ln -sf /opt/sdw/mailcap.default /home/user/.mailcap
+
+mkdir -p /home/user/.local/share/applications
+
+# XXX receive qube name (or its template in case of disposable) via arg due to
+# qubesdb not being able to obtain template name in case of disposable.
```

Comment:
Well...it would be nice if we could just have one systemd unit and this script read the VM name instead of hardcoding it in the service name. But you're saying that doesn't work for disposable VMs?

Reply:
> However, it worked for the first time and enabled all the system services in the `postinst`, but when building it and installing it again it stopped working. Do you know what could be happening? Does postinst/configure only run once?

Huh, not sure how it would've worked the first time, it's not supported by debhelper AFAICT. 

> but your approach may be simpler. @legoktm do you still recommend going down your suggestion 

That's still my preference but happy to be convinced otherwise.

> or do you have a hunch on how to get `postinst` enabling the required services?

We could do it manually, just have it call `systemctl enable mime-handling@sd-app`, etc. But you'll probably also want to do the same in postrm, etc. If you run `cat /var/lib/dpkg/info/securedrop-log.postinst` in a VM where it's installed, you can see how complicated the script generated by debehelper ends up being.

To list out all the options I see:

1. templated systemd units, with manual handling in postinst to enable them
2. duplicated systemd units that just vary on the VM name
3. single systemd unit, with a script that has logic to vary behavior on VM name.

My preference would be 3 > 2 > 1.